From cc2e0b1f5e54b26809aee001bf5c246ec20a788b Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Sun, 15 Dec 2024 07:47:30 +0200
Subject: [PATCH] [FSW-12567] fs1: remove invalid progress bits

Invalid flash offset progress bit is not related
to firmware image bringup sequence also unaligned
with enum boot_rom_bringup_proggress. Hence,
removed and adjusted bit sequence.

Change-Id: I6ad50ab1af5a1e46bdec7f5b6421961818ab825d
---

diff --git a/arch/fs1/arc/progress_bits.h b/arch/fs1/arc/progress_bits.h
index 2e8c194..fce49ce 100644
--- a/arch/fs1/arc/progress_bits.h
+++ b/arch/fs1/arc/progress_bits.h
@@ -83,79 +83,77 @@
 	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 73,
 	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 74,
 	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 75,
-	PROGRESS_BRINGUP_PCI_PRI_INVALID_FLASH_OFFSET = 76,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 77,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 78,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 79,
-	PROGRESS_BRINGUP_PCI_PRI_VERIFY_SUCCESS = 80,
-	PROGRESS_BRINGUP_PCI_SEC = 81,
-	PROGRESS_BRINGUP_PCI_SEC_NO_IMG_DESC = 82,
-	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID = 83,
-	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID = 84,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 85,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 86,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 87,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 88,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 89,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_FLASH_OFFSET = 90,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 91,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 92,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 93,
-	PROGRESS_BRINGUP_PCI_SEC_VERIFY_SUCCESS = 94,
-	PROGRESS_BOOT_ROM_TPM = 95,
-	PROGRESS_BOOT_ROM_TPM_INIT_FAIL = 96,
-	PROGRESS_BOOT_ROM_TPM_STARTUP_FAIL = 97,
-	PROGRESS_BOOT_ROM_SVN_TPM_PCR_EXTEND_FAIL = 98,
-	PROGRESS_BOOT_ROM_MANIFEST_TPM_PCR_EXTEND_FAIL = 99,
- 	PROGRESS_BOOT_ROM_TPM_FSM_OUT_OF_SYNC = 100,
-	PROGRESS_BOOT_ROM_TPM_FINISH = 101,
-	PROGRESS_PPBOOT_START = 102,
-	PROGRESS_PPBOOT_FINISH = 103,
-	PROGRESS_BOOT_ROM_IN_RECOVERY = 104,
-	PROGRESS_BOOT_ROM_IN_ROM_RECOVERY = 105,
-	PROGRESS_BOOT_ROM_PCIE_LINK_START = 106,
-	PROGRESS_BOOT_ROM_PCIE_LINK_INIT = 107,
-	PROGRESS_BOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 108,
-	PROGRESS_BOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 109,
-	PROGRESS_BOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 110,
-	PROGRESS_BOOT_ROM_PCIE_LINK_IATU_CONFIG = 111,
-	PROGRESS_BOOT_ROM_PCIE_LINK_FINISH = 112,
-	PROGRESS_BOOT_ROM_D2D_LINK_START = 113,
-	PROGRESS_BOOT_ROM_D2D_LINK_STS_BUSY = 114,
-	PROGRESS_BOOT_ROM_D2D_LINK_INTER_DIE_SYNC = 115,
-	PROGRESS_BOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 116,
-	PROGRESS_BOOT_ROM_D2D_LINK_INIT_SUCCESS = 117,
-	PROGRESS_BOOT_ROM_RECOVERY_END = 118,
-	PROGRESS_BOOT_ROM_IN_RECOVERY_IMG = 119,
-	PROGRESS_RECOVERY_IMG_IMG0 = 120,
-	PROGRESS_RECOVERY_IMG_IMG0_NO_IMG_DESC = 121,
-	PROGRESS_RECOVERY_IMG_IMG0_LOADADDR_INVALID = 122,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_STARTADDR_INVALID = 123,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_START = 124,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_CRC_FAILED = 125,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_UNALIGNED_DEST = 126,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_INVALID_IMG_SIZE = 127,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_UNALIGNED_IMG_SIZE = 128,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_INVALID_FLASH_OFFSET = 129,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_ERROR = 130,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_FINISH = 131,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_VERIFICATION_FAIL = 132,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_START = 133,
-	PROGRESS_BOOT_RECOVERY_IMG_IMG0_FINISH = 134,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_START = 135,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FINISH = 136,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_START = 137,
-	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FINISH = 138,
-	PROGRESS_BOOT_INFO_LFS_DISK_MOUNT_FAIL = 139,
-	PROGRESS_BOOT_INFO_LFS_FILE_OPEN_CFG_FAIL = 140,
-	PROGRESS_BOOT_INFO_LFS_FILE_COPY_PART_FAIL = 141,
-	PROGRESS_BOOT_INFO_LFS_FILE_SET_FAIL = 142,
-	PROGRESS_BOOT_INFO_LFS_FILE_NAME_FAIL = 143,
-	PROGRESS_BOOT_INFO_LFS_FINISHED = 144,
-	PROGRESS_BOOT_INFO_MANIFEST_CONTEXT_SET_FAIL = 145,
-	PROGRESS_BOOT_INFO_MANIFEST_HEADER_SET_FAIL = 146,
-	PROGRESS_BOOT_INFO_MANIFEST_CONTEXT_GET_FAIL = 147,
-	PROGRESS_BOOT_INFO_MANIFEST_HEADER_GET_FAIL = 148,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 76,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 77,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 78,
+	PROGRESS_BRINGUP_PCI_PRI_VERIFY_SUCCESS = 79,
+	PROGRESS_BRINGUP_PCI_SEC = 80,
+	PROGRESS_BRINGUP_PCI_SEC_NO_IMG_DESC = 81,
+	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID = 82,
+	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID = 83,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 84,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 85,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 86,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 87,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 88,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 89,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 90,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 91,
+	PROGRESS_BRINGUP_PCI_SEC_VERIFY_SUCCESS = 92,
+	PROGRESS_BOOT_ROM_TPM = 93,
+	PROGRESS_BOOT_ROM_TPM_INIT_FAIL = 94,
+	PROGRESS_BOOT_ROM_TPM_STARTUP_FAIL = 95,
+	PROGRESS_BOOT_ROM_SVN_TPM_PCR_EXTEND_FAIL = 96,
+	PROGRESS_BOOT_ROM_MANIFEST_TPM_PCR_EXTEND_FAIL = 97,
+ 	PROGRESS_BOOT_ROM_TPM_FSM_OUT_OF_SYNC = 98,
+	PROGRESS_BOOT_ROM_TPM_FINISH = 99,
+	PROGRESS_PPBOOT_START = 100,
+	PROGRESS_PPBOOT_FINISH = 101,
+	PROGRESS_BOOT_ROM_IN_RECOVERY = 102,
+	PROGRESS_BOOT_ROM_IN_ROM_RECOVERY = 103,
+	PROGRESS_BOOT_ROM_PCIE_LINK_START = 104,
+	PROGRESS_BOOT_ROM_PCIE_LINK_INIT = 105,
+	PROGRESS_BOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 106,
+	PROGRESS_BOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 107,
+	PROGRESS_BOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 108,
+	PROGRESS_BOOT_ROM_PCIE_LINK_IATU_CONFIG = 109,
+	PROGRESS_BOOT_ROM_PCIE_LINK_FINISH = 110,
+	PROGRESS_BOOT_ROM_D2D_LINK_START = 111,
+	PROGRESS_BOOT_ROM_D2D_LINK_STS_BUSY = 112,
+	PROGRESS_BOOT_ROM_D2D_LINK_INTER_DIE_SYNC = 113,
+	PROGRESS_BOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 114,
+	PROGRESS_BOOT_ROM_D2D_LINK_INIT_SUCCESS = 115,
+	PROGRESS_BOOT_ROM_RECOVERY_END = 116,
+	PROGRESS_BOOT_ROM_IN_RECOVERY_IMG = 117,
+	PROGRESS_RECOVERY_IMG_IMG0 = 118,
+	PROGRESS_RECOVERY_IMG_IMG0_NO_IMG_DESC = 119,
+	PROGRESS_RECOVERY_IMG_IMG0_LOADADDR_INVALID = 120,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_STARTADDR_INVALID = 121,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_START = 122,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_CRC_FAILED = 123,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_UNALIGNED_DEST = 124,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_INVALID_IMG_SIZE = 125,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_UNALIGNED_IMG_SIZE = 126,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_INVALID_FLASH_OFFSET = 127,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_ERROR = 128,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_DMA_FINISH = 129,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_VERIFICATION_FAIL = 130,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_START = 131,
+	PROGRESS_BOOT_RECOVERY_IMG_IMG0_FINISH = 132,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_START = 133,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FINISH = 134,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_START = 135,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FINISH = 136,
+	PROGRESS_BOOT_INFO_LFS_DISK_MOUNT_FAIL = 137,
+	PROGRESS_BOOT_INFO_LFS_FILE_OPEN_CFG_FAIL = 138,
+	PROGRESS_BOOT_INFO_LFS_FILE_COPY_PART_FAIL = 139,
+	PROGRESS_BOOT_INFO_LFS_FILE_SET_FAIL = 140,
+	PROGRESS_BOOT_INFO_LFS_FILE_NAME_FAIL = 141,
+	PROGRESS_BOOT_INFO_LFS_FINISHED = 142,
+	PROGRESS_BOOT_INFO_MANIFEST_CONTEXT_SET_FAIL = 143,
+	PROGRESS_BOOT_INFO_MANIFEST_HEADER_SET_FAIL = 144,
+	PROGRESS_BOOT_INFO_MANIFEST_CONTEXT_GET_FAIL = 145,
+	PROGRESS_BOOT_INFO_MANIFEST_HEADER_GET_FAIL = 146,
 };
 
 #endif /* _PROGRESS_BITS_H_ */
